libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -30,28 +62,54 @@ void VotePacketHandler::process(const PacketData &packet_data, const std::shared
     const auto vote_hash = vote->getHash();
     LOG(log_dg_) << "Received PBFT vote " << vote_hash;
 
+    // Check period of vote. Request PBFT sync if we have a future one here.
+    if (vote->getPeriod() > pbft_chain_->getPbftChainSize() + kVoteAcceptingPeriods) {
+      LOG(log_er_) << "Skip vote " << vote_hash.abridged() << ". Period is too big " << pbft_chain_->getPbftChainSize()
+                   << " " << vote->getPeriod();
+      // Do not request round sync too often here
+      if (std::chrono::system_clock::now() - round_sync_request_time_ > kSyncRequestInterval) {
+        // request sync from this node
+        sealAndSend(peer->getId(), SubprotocolPacketType::GetPbftSyncPacket,
+                    std::move(dev::RLPStream(1) << vote->getPeriod()));
+        round_sync_request_time_ = std::chrono::system_clock::now();
+      }
+      continue;
+    }
+    // Check this before we are marking vote as seen to have an opportunity to process it after
+    if (!isVoteRoundStepRelevant(vote)) {
```

Comment:
this should be called only for standard vote as it does not make sense for reward vote

Reply:
also this is checking/validating vote max round & step, so I guess something like validateVoteMaxRoundAndStep would be more clear